packages/lesswrong/components/posts/AddToCalendar/AddToCalendarIcon.tsx

```diff
@@ -57,7 +57,7 @@ const AddToCalendarIcon = ({post, label, hideTooltip, classes}: {
         event={{
           name: post.title,
           details: eventDetails,
-          location: post.location,
+          location: post.onlineEvent ? post.joinEventLink : post.location,
```

Comment:
In my commit just now, I slightly updated the event time format on the event details page and put the `joinEventLink` in the location field of "add to calendar".

Also, now we display the `joinEventLink` URL (truncated) on the event details page under the event time, and the "Join Event" button is disabled until 10 min before the event starts.

<img width="746" alt="Screen Shot 2022-01-12 at 3 15 04 PM" src="https://user-images.githubusercontent.com/9057804/149217686-b84bea35-7e28-486b-a78d-ce60d836b0b5.png">